account_invoice_date_due/models/account_move.py

```diff
@@ -0,0 +1,30 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    # Show invoice date due even when payment term is defined
+    invoice_date_due_payment_term = fields.Date(
+        related="invoice_date_due", string="Due Date Payment Term"
+    )
+
+    @api.onchange("invoice_date_due_payment_term")
+    def _onchange_invoice_date_due_payment_term(self):
+        """Propagate from Payment term due date to original field"""
+        if self.invoice_date_due_payment_term:
+            self.invoice_date_due = self.invoice_date_due_payment_term
+
+    def write(self, vals):
+        res = super().write(vals)
+        # Propagate due date to move lines
+        # that correspont to the receivable/payable account
+        if "invoice_date_due" in vals and self.state in ("draft", "posted"):
```

Comment:
I'd say you won't have payment term lines when the invoice is in draft
```suggestion
        if "invoice_date_due" in vals and self.state == "posted":
```

Reply:
Yes, it was just that, from my testing, the correspondent move lines that should be updated are created already while in draft mode. 
Should I just ignore the values while in draft and change only when posted?

Reply:
The thing is that when you're in draft you'll work with the standard behavior. The edition of the due date should be allowed after posting the invoice

Reply:
Ok, I think I was looking at the problem wrong :sweat_smile: 
I removed that condition following your suggestion and also removed the check from the tests (I now only check the propagation after having posted the invoice)